aws/middleware/header.go

```diff
@@ -0,0 +1,82 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+)
+
+// WithHeaderComment instruments a middleware stack to append an HTTP field
+// comment to the given header as specified in RFC 9110
+// (https://www.rfc-editor.org/rfc/rfc9110#name-comments).
+//
+// The header is case-insensitive. If the provided header exists when the
+// middleware runs, the content will be inserted as-is enclosed in parentheses.
+//
+// Note that per the HTTP specification, comments are only allowed in fields
+// containing "comment" as part of their field value definition, but this API
+// will NOT verify whether the provided header is one of them.
+//
+// WithHeaderComment MAY be applied more than once to a middleware stack and/or
+// more than once per header.
+func WithHeaderComment(header, content string) func(*middleware.Stack) error {
+	return func(s *middleware.Stack) error {
+		m, err := getOrAddHeaderComment(s)
```

Comment:
**Question**: Does this get wired up by default in a way that runs after the UserAgent middleware?

Reply:
It's not there by default at all, so if you add it, it's guaranteed to come after.

Reply:
Sorry not what I meant, I meant that if you add this option/middleware is it (by default) after the UA middleware? If it's not then this would have no effect on the UA. I think the answer is yes but I didn't actually go verify.

Reply:
Correct - confirmed offline